tabular/tests/unittests/resource_allocation/test_total_resource_allocation.py

```diff
@@ -0,0 +1,232 @@
+import pytest
+import pandas as pd
+
+from autogluon.core.utils import get_cpu_count, get_gpu_count_all
+from autogluon.core.models.ensemble.bagged_ensemble_model import BaggedEnsembleModel
+from autogluon.tabular.models import AbstractModel
+
+
+class DummyBaseModel(AbstractModel):
+    
+    def __init__(self, minimum_resources={}, **kwargs):
+        self._minimum_resources = minimum_resources
+        super().__init__(**kwargs)
+        
+    def get_minimum_resources(self, **kwargs):
+        return self._minimum_resources
+    
+    def _get_default_resources(self):
+        num_cpus = 1
+        num_gpus = 1
+        return num_cpus, num_gpus
+
+
+class DummyModel(DummyBaseModel):
+    pass
+
+    
+class DummyBaggedModel(BaggedEnsembleModel):
+    pass
+
+
+get_cpu_count.mock = 16
+get_gpu_count_all.mock = 2
```

Comment:
1. Will this alter other tests?
2. Assign to variables: MOCK_CPU_COUNT = 16, MOCK_GPU_COUNT = 2, can re-use in tests

Reply:
I've added a context manager within conftest and fixtures to pass in the context manager and mock variables to tests